go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -223,27 +223,17 @@ func (oc *Controller) addEgressFirewall(egressFirewall *egressfirewallapi.Egress
 		return fmt.Errorf("cannot Ensure that addressSet for namespace %s exists %v", egressFirewall.Namespace, err)
 	}
 	ipv4HashedAS, ipv6HashedAS := addressset.MakeAddressSetHashNames(egressFirewall.Namespace)
-	err = oc.addEgressFirewallRules(ef, ipv4HashedAS, ipv6HashedAS, types.EgressFirewallStartPriority)
-	if err != nil {
+	if err := oc.addEgressFirewallRules(ef, ipv4HashedAS, ipv6HashedAS, types.EgressFirewallStartPriority); err != nil {
 		return err
 	}
-
+	oc.egressFirewalls.Store(egressFirewall.Namespace, ef)
 	return nil
 }
 
-func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall *egressfirewallapi.EgressFirewall) error {
-	updateErrors := oc.deleteEgressFirewall(oldEgressFirewall)
-	if updateErrors != nil {
-		return updateErrors
-	}
-	updateErrors = oc.addEgressFirewall(newEgressFirewall)
-	return updateErrors
-}
-
 func (oc *Controller) deleteEgressFirewall(egressFirewallObj *egressfirewallapi.EgressFirewall) error {
 	klog.Infof("Deleting egress Firewall %s in namespace %s", egressFirewallObj.Name, egressFirewallObj.Namespace)
 	deleteDNS := false
-	obj, loaded := oc.egressFirewalls.LoadAndDelete(egressFirewallObj.Namespace)
+	obj, loaded := oc.egressFirewalls.Load(egressFirewallObj.Namespace)
 	if !loaded {
 		return fmt.Errorf("there is no egressFirewall found in namespace %s",
 			egressFirewallObj.Namespace)
```

Comment:
Hmm... should we consider it an error if we delete an egress firewall that doesn't exist? If we do so, then we'll retry deleting it forever. Can we log the message as a warning and return nil?

Reply:
How can we get delete event for something that doesn’t exists it’s an error probably retry can’t help it but it’s an error still ?

Reply:
@msherif1234 I would do this in `iterateRetryResources` on a per-entry basis (e.g. pod X) instead of an per-resource basis (e.g. all pods). Let me take care of that in a separate PR. @trozet, what do you think? Does it make sense to you to put an upper bound on the max number of retries of a single object? If so, any suggestion on what this upper bound should be (100? 1000? more?)

Reply:
It looks like most of the uses of the object where taken out but there still is one. The sync map is still used to determine if there is already an egressFirewall applied to the current namespace

Reply:
@ricky-rav in the level driven world we requeue a certain number of times and then give up. We should abandon if the resource is gone or some max number of tries I think

Reply:
yeah we did something like this for level driven
```golang
const (
    maxNodeRetries = 10
)
......
        // since we failed, we should requeue the item to work on later.
	// but only if we've not exceeded max retries. This method
	// will add a backoff to avoid hotlooping on particular items
	// (they're probably still not going to work right away) and overall
	// controller protection (everything I've done is broken, this controller
	// needs to calm down or it can starve other useful work) cases.
	if oc.nodesQueue.NumRequeues(key) < maxNodeRetries {
		oc.nodesQueue.AddRateLimited(key)
		return true
	}
```